src/org/thoughtcrime/securesms/WebxdcActivity.java

```diff
@@ -266,18 +266,16 @@ private void setScreenMode(Configuration config) {
     }
   }
 
+  // This is only called when internetAccess == true or for mailto/openpgp4fpr scheme,
+  // because when internetAccess == false, the page is loaded inside an iframe,
+  // and WebViewClient.shouldOverrideUrlLoading is not called for HTTP(S) links inside the iframe
   @Override
   protected boolean openOnlineUrl(String url) {
-    if (internetAccess) {
-      // internet access enabled, continue loading in the WebView
-      return false;
-    }
-    if (url.startsWith("mailto:")) {
+    Log.i(TAG, "openOnlineUrl: " + url);
+    if (url.startsWith("mailto:") || url.startsWith("openpgp4fpr:")) {
       return super.openOnlineUrl(url);
     }
-
-    Toast.makeText(this, "Please embed needed resources.", Toast.LENGTH_LONG).show();
-    return true; // returning `true` causes the WebView to abort loading
+    return false; // continue loading in the WebView
```

Comment:
i find this too dangerous and implicit. might be, that we get rid of iframes at some point (hopefully) or flow changes otherwise - it is so easy to forget this bit then.

therefore, and also as another defense line, let's stay with the `internetAccess` check:

```suggestion
    return !internetAccess; // returning `false` continues loading in WebView; returning `true` let WebView abort loading
```

removing the toast is still fine, as this is not the happy path.

Reply:
notice that continuing loading only means that the page is loaded in place which is then blocked by CSP and if that barrier would fail somehow then it would be blocked by the fake-proxy,

but I will merge it as a 3rd layer shield :o